indico/web/client/js/react/components/principals/PrincipalListField.jsx

```diff
@@ -44,6 +45,7 @@ const PrincipalListField = props => {
     eventId,
     favoriteUsersController,
     className,
+    fieldId,
```

Comment:
Can we somehow avoid that? When using this outside a WTForm (typically using `FinalPrincipalList`), there is no hidden field with an ID that could be used by the code connecting to your signal...

Reply:
I passed the `fieldId` for a `boolean` set on the `PrincipalListField` so that we can avoid this, so I had to make some changes on how to generate the CSV export file; now created on the front-end instead of the back-end

Reply:
I'm not sure if that's better. The whole idea of plugin interfaces is that they are generic. `importExport` is rather specific to what this particular plugin does.

I also don't have a better idea right now though (but TBH I prefer passing the ID over this importExport boolean), but I'll see if I can come up w/ something...

Reply:
IIUC there are two things we need to pass (for this specific feature):
 - whether the button is for import or export (`importExport`)
 - if it is an export button, whether it is for judges or reviewers (we get this information from the `fieldId`)

The `fieldId` is needed only for the second one, but we discussed with Michel and we can probably remove it. This still leaves us with the the `importExport` prop though

Reply:
I passed the fieldId before as I could use it for both, determine if the buttons should be rendered (only for the `fieldId`s of `judges` and `content_reviewers`) and to ask for the corresponding list of users for both fields when exporting.

By generating the CSV on the front end I don't need the fieldId to determine which users to include in the exported file (since they are state variables within each PrincipalListField` component  individually), so instead I added the `importExport` prop that just tells the `PrincipalListField` if it should render the buttons for `import` AND `export` or not.

Without any flag like the `importExport` boolean getting the buttons to rendered only on the desired `PrincipalListFiled`s is not possible, and maybe the `fieldId` might seem to specific for the current fields targeted.

Reply:
@ThiefMaster any news on this, should I just rollback to the `fieldId` solution?

Reply:
I'll have a look once I'm done w/ the event/category import/export stuff... still not clear what's the best solution there, but I think when checking out the branch and trying things out it'll become clearer than from just looking at the diffs here

Reply:
So I still don't like this at all, because those field ids are not even guaranteed to be unique across pages (only within a single page). I got an idea how we could do it much more nicely using react context. Will add a commit to both PRs in a bit if it works as expected.